src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java

```diff
@@ -183,6 +189,29 @@ private Optional<ProjectBindingWrapper> getBinding(Optional<WorkspaceFolderWrapp
     });
   }
 
+  public CompletableFuture<String> getUpdatedBindingForWorkspaceFolder(URI folderUri) {
+    var countDown = new CountDownLatch(1);
```

Comment:
I would maybe rename this variable to `bindingUpdatedLatch` to make it more explicit what we are waiting for.